Revit_Core_Engine/Convert/Architecture/FromRevit/Opening.cs

```diff
@@ -0,0 +1,112 @@
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("opening", "Architecture.BuildersWork.Opening resulting from converting the input Revit FamilyInstance.")]
+        public static oM.Architecture.BuildersWork.Opening OpeningFromRevit(this FamilyInstance instance, RevitSettings settings = null, Dictionary<string, List<IBHoMObject>> refObjects = null)
+        {
+            if (instance == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError($"The convert from Revit element of type {typeof(FamilyInstance).Name} to {typeof(oM.Architecture.BuildersWork.Opening).Name} failed because the input element is null.");
+                return null;
+            }
+
+            settings = settings.DefaultIfNull();
+
+            oM.Architecture.BuildersWork.Opening opening = refObjects.GetValue<oM.Architecture.BuildersWork.Opening>(instance.Id);
+            if (opening != null)
+                return opening;
+            
+            opening = new oM.Architecture.BuildersWork.Opening { Name = instance.Name };
+
+            //Set coordinate system
+            opening.CoordinateSystem = instance.CoordinateSystem();
+
+            IProfile profile = null;
+            Parameter diameterParam = instance.LookupParameter(settings.ParameterSettings, typeof(oM.Architecture.BuildersWork.Opening), nameof(CircleProfile.Diameter));
+            if (diameterParam != null)
+            {
+                double diameter = diameterParam.AsDouble().ToSI(diameterParam.Definition.UnitType);
+                profile = BH.Engine.Spatial.Create.CircleProfile(diameter);
+            }
+
+            if (profile == null)
+            {
+                Parameter widthParam = instance.LookupParameter(settings.ParameterSettings, typeof(oM.Architecture.BuildersWork.Opening), nameof(RectangleProfile.Width));
+                Parameter heightParam = instance.LookupParameter(settings.ParameterSettings, typeof(oM.Architecture.BuildersWork.Opening), nameof(RectangleProfile.Height));
+                if (widthParam != null && heightParam != null)
+                {
+                    double width = widthParam.AsDouble().ToSI(widthParam.Definition.UnitType);
+                    double height = heightParam.AsDouble().ToSI(heightParam.Definition.UnitType);
+                    profile = BH.Engine.Spatial.Create.RectangleProfile(height, width);
+                }
+            }
+
+            if (profile == null)
+                BH.Engine.Reflection.Compute.RecordWarning($"The profile of the opening could not be extracted from the Revit element because the parameters correspondent to their dimensions could not be found. Revit ElementId: {instance.Id}"
+                    + "\nTo link the specific parameter values with opening height/width/diameter, add relevant ParameterMap to RevitSettings.ParameterSettings.");
+
```

Comment:
I always found it hard to embed Wiki links in the messages because one can't click on them. On the other hand, the link can be copied or simply browsed, so I a agree, makes sense. Issue raised: https://github.com/BHoM/Revit_Toolkit/issues/1061